src/progress_bar_exercise/components/ProgressBar/ProgressBar.js

```diff
@@ -0,0 +1,56 @@
+import React, { useEffect, useMemo } from "react";
+import "./ProgressBar.scss";
+import { DEFAULT_SECONDS, useProgressBarState } from "./useProgressBarState";
+
+// Normally we would be using prop-types or typescript to type-check our props
+const ProgressBar = ({ loading, expectedSeconds = DEFAULT_SECONDS }) => {
+  const {
+    hasStarted,
+    transitionDuration,
+    stopMilliseconds,
+    animatedProgress,
+    addNormalAnimation,
+    handleDone,
+  } = useProgressBarState(expectedSeconds);
+
+  useEffect(() => {
+    if (loading) {
+      addNormalAnimation(stopMilliseconds);
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
**Note for reviewers:**

Here we are disabling the dependencies because `addNormalAnimation` and `stopMilliseconds` change often, making this effect be called many times, so it's easier in the case to just pass one dependency